sycl/doc/OptionalDeviceFeatures.md

```diff
@@ -629,11 +629,122 @@ example, if a new device is released before there is a new DPC++ release.  In
+below illustrates the action graph built by the clang driver along with file
+lists and tables generated and consumed by various nodes of the graph. The
+example set of targets used for the illustration is 4 targets
+- spir64 (runtime JITted SPIR-V)
+- AOT targets
+    - non-SPIR-V based
+        - nvptx (PTX)
+    - SPIR-V based
+        - gen_11 (Intel Gen)
+        - x86_64_avx512 (AVX512)
+<br>
+<br>
+
+![Device SPIRV translation and AOT compilation](images/DeviceLinkAOTAndWrap.svg)
+
+##### Aspect filter tool
+
+This tool transforms an input file table by removing rows with device code files
+that use features unsupported for the target architecture given as tool's
+argument.
+
+*Name*:
+- `sycl-aspect-filter`, located next to other tools like `file-table-tform`
+
+*Input*:
+- file table, normally coming out of `sycl-post-link` or `file-table-tform`
+  tools
+
+*Command line arguments*
+- `-target=<target>` target device architecture to filter for
+- `-device-config-file=<path>` path to the device configuration file
+
+*Output*
+- the input file table filtered as needed
+
+In more details, the tool performs the following actions:
+1) Checks if the input file table contains "Properties" column. If not, copies
+   the input file table to output and exits without error.
+1) Reads in the device configuration file and finds some entry `E` corresponding
+   to the architecture given on the command line. If there is no such entry -
+   reports and error and exits.
+1) For each row in the input file table:
+   - loads the properties file from the "Properties" column
+   - checks if there is the `SYCL/image-requirements` property
+   - if no, copies current row to the output file table and goes to the next
+   - if yes, checks if all the requirements listed in the property are supported
+     by the target architecture as specified by entry `E` in the device configuration file
+       - if yes, copies current row to the output file table and goes to the
+         next
+       - otherwise skips this row
+
+##### Configuration file location and driver option
+
+A default device configuration file is supplied as a part of OneAPI SDK. It is
+located in the TBD directory. Users may override the defalt using the
+```
+-fsycl-device-config-file=<path>
+```
+option.
+
+##### AOT target identification
+
+There are several user-visible places in the SDK where SYCL device target
+architectures need to be identified:
+- `-fsycl-targets` option
+- a device configuration file entry
+- `-target` option of the `sycl-aspec-filter` tool
+- a SYCL aspect enum identifier
 
-[7]: <https://github.com/gmlueck/llvm/pull/1>
+In all such places architecture naming should be the same. In some cases aliases
+are allowed. Below is a list of target architectures supported by DPC++:
 
+| target/alias(es) | description |
+|-|-|
+| fpga   | Generic Intel FPGA accelerator architecture |
+| intel_gpu    | Generic Intel graphics architecture |
```

Comment:
I'm not sure about these first two entries because they don't identify a specific target device.  Will it be legal to AOT-compile with `-fsycl-targets=fpga` or `-fsycl-targets=intel_gpu`?  Don't you need to name a specific target device instead?

Reply:
Yes, they should be legal and reflect some generic device architecture, such that code compiled with this target architecture will run on any device of this kind. In fact, intel_gpu might mean generating code for every arch supported by `ocloc`.  BTW, `fpga` AOT target already exists, so we need to support it. See e.g. https://github.com/intel/llvm-test-suite/blob/intel/SYCL/AOT/accelerator.cpp